src/app/controllers/main.controller.ts

```diff
@@ -1,79 +1,40 @@
 module webShoes {
   'use strict';
 
-  class Thing {
-    public rank: number;
-    public login: string;
-    public avatar_url: string;
-    public html_url: string;
-
-    constructor(login: string, avatar_url: string, rank: number, html_url: string) {
-      this.login = login;
-      this.avatar_url = avatar_url;
-      this.html_url = html_url;
-
-    }
+  class Product {
+    id         : number;
+    img1       : string;
+    img2       : string;
+    img3       : string;
+    img4       : string;
+    manufactor : string;
+    name       : string;
+    price      : number;
+    size       : string;
   }
 
   export class MainController {
-    public awesomeThings: Thing[];
-    public search: string;
-    private pendingTask: any;
-    private $http: any;
-
-    updateGrid(response) {
-      var _this = this;
-        this.awesomeThings = new Array<Thing>();
-        _.forEach(response.data, function(awesomeThing: Thing) {
-            _this.awesomeThings.push(awesomeThing);
-        });
-    }
-
-    getAllUsers() {
-        var _this = this;
-        var data;
-        this.$http.get('https://api.github.com/users')
-            .then(function(response){_this.updateGrid(response);});
-    }
-
-    fetchUser() {
-        var awesomeThings;
-        var _this = this;
-        if (this.search.length <= 0) {
-            this.getAllUsers();
-            return;
-        }
-        this.$http.get('https://api.github.com/search/users?q=' + this.search)
-            .then(function(response) {
-                awesomeThings = response.data.items;
-                _this.awesomeThings = new Array<Thing>();
-
-                awesomeThings.forEach(function(profile: Thing) {
-                  _this.awesomeThings.push(profile);
-
-                });
-        });
-    }
-
-    change() {
-      if (this.pendingTask) {
-        clearTimeout(this.pendingTask);
+    public $http: any;
+    public $window: any;
+    public rootUrl: string = 'http://webshoes-backend.herokuapp.com';
+    public productList : Array<Product>;
+
+      constructor ($scope: any, $http: any, $window: any) {
+        this.$http = $http;
+        this.$window = $window;
+        this.onGetProducList();
```

Comment:
@lucasBgabriel onGetProductList() is more of a event function, in this case I think only getProductList() would be better.